File: backend/app/schemas/hosted_zone.py
from datetime import datetime
from pydantic import BaseModel


class HostedZoneCreate(BaseModel):
    name: str
    type: str = "Public"
    comment: str = ""


class HostedZoneUpdate(BaseModel):
    comment: str | None = None
    status: str | None = None


class HostedZoneResponse(BaseModel):
    id: int
    zone_id: str
    name: str
    type: str
    comment: str
    status: str
    name_servers: str  # JSON string of nameservers
    record_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class HostedZoneListResponse(BaseModel):
    items: list[HostedZoneResponse]
    page: int
    limit: int
    total: int
    total_pages: int

File: frontend/src/types/hosted-zone.ts
export interface HostedZone {
  id: number;
  zone_id: string;
  name: string;
  type: 'Public' | 'Private';
  comment: string;
  status: 'Active' | 'Inactive';
  name_servers: string; // JSON string array
  record_count: number;
  created_at: string;
  updated_at: string;
}

export interface HostedZoneCreateInput {
  name: string;
  type: 'Public' | 'Private';
  comment?: string;
}

export interface HostedZoneUpdateInput {
  comment?: string;
  status?: 'Active' | 'Inactive';
}

export interface PaginatedResponse<T> {
  items: T[];
  page: number;
  limit: number;
  total: number;
  total_pages: number;
}
